src/main/java/build/buildfarm/common/config/GrpcMetrics.java

```diff
@@ -30,6 +32,11 @@ public static void handleGrpcMetricIntercepts(
         grpcConfig = grpcConfig.withLatencyBuckets(grpcMetrics.getLatencyBuckets());
       }
 
+      // report custom metric labels
+      if (grpcMetrics.getLabelsToReport() != null && !grpcMetrics.getLabelsToReport().isEmpty()) {
+        grpcConfig = grpcConfig.withLabelHeaders(grpcMetrics.getLabelsToReport());
+      }
+
```

Comment:
Shouldn't need the empty check here - we don't do it on the latency buckets above.

Reply:
Removed. I put the check there to avoid unknown behaviour from passing empty label arrays to grpcConfig.withLabelHeaders, since I did not look at the implementation of that function to ensure it is not an issue.

Reply:
I took a look at it. Good consideration, although the prior art above with existing list specification, and general API expectations are that adding all elements of an empty list should noop and represent no difference to not calling the add in the first place.